web/src/enterprise/campaigns/detail/AddChangesetForm.tsx

```diff
@@ -0,0 +1,115 @@
+            gql`
+                query RepositoryID($repoName: String!) {
+                    repository(name: $repoName) {
+                        id
+                    }
+                }
+            `,
+            { repoName }
+        ).toPromise()
+    ).repository
+    if (!repository) {
+        throw new RepoNotFoundError(repoName)
+    }
+
+    const changeset = dataOrThrowErrors(
+        await mutateGraphQL(
+            gql`
+                mutation CreateChangeSet($repositoryID: ID!, $externalID: String!) {
+                    createChangesets(input: { repository: $repositoryID, externalID: $externalID }) {
+                        id
+                    }
+                }
+            `,
+            { repositoryID: repository.id, externalID }
+        ).toPromise()
+    ).createChangesets[0]
+
+    dataOrThrowErrors(
+        await mutateGraphQL(
+            gql`
+                mutation AddChangeSetToCampaign($campaignID: ID!, $changesets: [ID!]!) {
+                    addChangesetsToCampaign(campaign: $campaignID, changesets: $changesets) {
+                        id
+                    }
+                }
+            `,
+            { campaignID, changesets: [changeset.id] }
+        ).toPromise()
+    )
+}
+
+/**
+ * Simple, temporary form to add changesets.
+ */
+export const AddChangesetForm: React.FunctionComponent<{ campaignID: ID; onAdd: () => void }> = ({
+    campaignID,
+    onAdd,
+}) => {
+    const triggerError = useError()
+    const [repoName, setRepoName] = useState('')
+    const [externalID, setExternalID] = useState('')
+    const [isLoading, setIsLoading] = useState(false)
+    const submit = useCallback(
+        async (event: FormEvent) => {
+            event.preventDefault()
+            try {
+                setIsLoading(true)
+                await addChangeset({ campaignID, repoName, externalID })
+                setExternalID('')
+                onAdd()
+            } catch (err) {
+                triggerError(err)
+            } finally {
+                setIsLoading(false)
+            }
+        },
+        [campaignID, externalID, onAdd, repoName, triggerError]
+    )
+    return (
+        <Form className="form-inline" onSubmit={submit}>
+            <input
+                required={true}
+                type="text"
+                size={35}
+                className="form-control mr-1"
+                placeholder="Repository name"
+                value={repoName}
+                onChange={event => setRepoName(event.target.value)}
+            />
+            <input
+                required={true}
+                type="text"
+                size={16}
+                className="form-control mr-1"
+                placeholder="Changeset number"
+                value={externalID}
+                onChange={event => setExternalID(event.target.value)}
+            />
+            <button type="submit" className="btn btn-primary mr-1">
+                Add Changeset
```

Comment:
```suggestion
                Add changeset
```

[Use sentence case](https://docs.sourcegraph.com/dev/style_guide#general).